frontend/public/kubevirt/components/vm.jsx

```diff
@@ -330,6 +230,56 @@ const mapDispatchToProps = () => ({
   watchK8sList: actions.watchK8sList,
 });
 
+const ConnectedVmDetails = ({ obj: vm }) => {
+  const vmResource = getResourceKind(VirtualMachineModel, vm.metadata.name, true, vm.metadata.namespace, false);
+  const vmiResources = getResourceKind(VirtualMachineInstanceModel, vm.metadata.name, true, vm.metadata.namespace, true, getLabelMatcher(vm));
+  const podResources = getResourceKind(PodModel, undefined, true, vm.metadata.namespace, true, getLabelMatcher(vm));
+  const migrationResources = getResourceKind(VirtualMachineInstanceMigrationModel, undefined, true, vm.metadata.namespace, false); // https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachineinstancemigration
+  const resources = [vmResource, vmiResources, podResources, migrationResources];
+
+  return (
+    <Firehose resources={resources} flatten={getFlattenForKind(VirtualMachineModel.kind)}>
+      <VmDetails_ vm={vm} />
+    </Firehose>
+  );
+};
+
+const VmDetails_ = props => {
+  const { flatten, loaded, vm, resources } = props;
+  if (loaded){
+    const vm = flatten(props.resources);
+
+    if (vm) {
+      const namespaceResourceLink = (
+        <ResourceLink kind={NamespaceModel.kind} name={vm.metadata.namespace} title={vm.metadata.namespace} />
+      );
+
+      const podResourceLink = (
+        <FirehoseResourceLink
+          {...props}
+          flatten={getFlattenForKind(PodModel.kind)}
+          filter={data => findPod(data, vm.metadata.name, VIRT_LAUNCHER_POD_PREFIX)}
+        />);
+
+      return (
+        <VmDetails
```

Comment:
since we are watching multiple resources with firehose, each of them has its own loaded prop. It is important to check loadedError too. I suggest to create a new helper method
```
const getLoadedResource = (resource, defaultValue = []) => {
  if (!resource.loaded) {
    return defaultValue;
  }
  return resource.loadError ? defaultValue : resource.data;
};
```

and use it in this context as


```
const pods = getLoadedResource(resources[PodModel.kind]);
const migrations = getLoadedResource(resources[VirtualMachineInstanceMigrationModel.kind]);
// since VMI is not a list but a single object we want null as default return value if not yet loaded or there was an error while loading
const vmi = getLoadedResource(resources[VirtualMachineInstanceModel.kind], null);
const migration = vmi ? findVMIMigration(migrations, vmi.metadata.name) : null;
```

and then pass the `pods, migration, vmi` to `<VmDetails>`